### ISSUE TITLE
high validation error when training a imageNet model with version beta17

### ISSUE THREAD

**Issue by okvol** [vlfeat/matconvnet#365]
![high_validation_error](https://cloud.githubusercontent.com/assets/13367977/12036952/acd63ad0-ae18-11e5-939a-e961dfaa2c2d.png)

Often I have unusually high validation error with training imageNet model with beta17.
Does anybody else have this problem?

So far it's only the problem with ImageNet. I haven't observed this problem with MNIST or CIFAR.

**Comment by tinalegre** [comment]
@ovkol this is strange. I trained AlexNet on Imagenet using v17.0 without any problems (networktype: `simplenn` not `dagnn`). Have a look at the obtained [learning curve](goo.gl/hI9xfI) from my model. Did you change any default parameter?

**Comment by rosenfeldamir** [comment]
@okvol , I've been having the same issue. I think this has to do with a few CMYK images in the imagenet dataset. After having converted all of the CMYK images to RGB (using ImageMagick), the issue seems to have been solved. 
Note that there are very few (around 10, maybe a bit more) such images in the dataset and it seems unlikely that this will affect the training so much. I'll write about it when I have a clearer answer. See also here : #91 .
@tinalegre  - did you preprocess the images before you started training? Did you enounter any CMYK issues ?

**Comment by tinalegre** [comment]
@rosenfeldamir yes, I basically followed the suggestion of the post you mentioned on CMYK images. You can find my cnn_imagenet_get_batch.m adapted function in this [link](http://paste.ofcode.org/PQfDEUmURDcsBmEYKjSqAp).  I hope this can help you. Pls, keep me updated ;-)
On the other hand, I updated the link on my previous post on the obtained learning curve. I noticed, the link was broken.

**Comment by rosenfeldamir** [comment]
@tinalegre , I think you should look at #91 : tl;dr : you're training using a few badly loaded (CMYK) images. Overall this has no effect - but you can solve it by converting them using an external tool, e.g., ImageMagick.

**Comment by okvol** [comment]
I found that the error was caused by "Inf" values in one of the batch normalization modules.
namely all values in net.params(5).value(:,2) are Inf. This caused the test performance to be at chance.
I don't know why they became Inf in the first place. Maybe caused by the corrupted training images?
I'll keep you guys updated.

**Comment by rosenfeldamir** [comment]
@okvol have a look at #91. The "inf" values an after-effect of vl_imreadjpeg not reading some images correctly , namely CMYK images.

**Comment by vedaldi** [comment]
Hi, I strongly recommend pre-processing the ImageNet images converting the color space (As well as resizing the images; the speedup can be substantial). There is a limitation in Linux’s libjpeg (no CMYK -> RGB conversion) and we would have to implement our own conversion code to work around this.

The next version (on Linux) will print warning messages and refuse to read CMYK images to make this explicit.
